content/docs/reference/state.md

```diff
@@ -1,140 +1,212 @@
+and security for most users. But you can opt into the filesystem or cloud storage backend for more flexibility if you need it.
+
+
+<table>
+    <colgroup>
+        <col style="width: 20%">
+        <col style="width: 20%">
+        <col style="width: 35%">
+    </colgroup>
+        <thead>
+                <th>Backend</th>
+                <th>Description</th>
+                <th>Setup</th>
+        </thead>
+        <tbody>
+            <tr>
+                <td><code>pulumi.com</code></td>
+                <td> Accessed by the CLI through REST API calls</td>
+                <td>No additional configuration after <a href="{{< relref "/docs/reference/install" >}}" target="_blank">installing Pulumi.</a></td>
+            </tr>
+            <tr>
+                <td>Self-managed</td>
+                <td>Stored locally on your computer’s filesystem, or remotely using a cloud provider</td>
+                <td>
+                    <ul>
+                        <li>
+                        <b>Filesystem or local</b>: <i>See <a href="{{< relref "/docs/reference/clouds/aws/setup" >}}" target="_blank">AWS Setup</a> and <a href="https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingBucket.html" target="_blank">Working with Amazon S3 buckets</a> for configuration details.</i>
+                        </li>
+                        <li>
+                        <b>Cloud storage or remote</b>
+                            <ul class="p-0 list-none">
+                                <li>
+                                <b>AWS S3:</b> <i>See <a href="{{< relref "/docs/reference/clouds/aws/setup" >}}" target="_blank">AWS Setup</a> and <a href="https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingBucket.html" target="_blank">Working with Amazon S3 buckets</a> for configuration details.</i>
+                                </li>
+                                <li>
+                                <b>Azure Blob:</b> <i>See <a href="{{< relref "/docs/reference/clouds/azure/setup" >}}" target="_blank">Azure Setup</a> and Microsoft's <a href="https://docs.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-cli" target="_blank">Storage Blobs Quickstart</a> for configuration details.</i>
+                                </li>
+                                <li>
+                                <b>Google Cloud Storage:</b> <i>See <a href="{{< relref "/docs/reference/clouds/gcp/setup" >}}" target="_blank">GCP Setup</a> and Google's <a href="https://cloud.google.com/storage/docs/quickstarts" target="_blank">Cloud Storage Quickstarts</a> for configuration details.
+                                </li>
```

Comment:
i think we should first explain what it means to store the checkpoint "remotely using a cloud provider" before getting into the quickstarts on how to use their particular backends.  once we explain it, we can start split into the three tracks.

Reply:
Hey @CyrusNajmabadi - is [this](https://github.com/pulumi/docs/pull/1390#discussion_r306097412) along the lines of what you're looking for? The anchor links to the local and remote "login" options do go into detail about where the files get stored. I have a follow-up question around backups however.